drivers/lxc/lxc.go

```diff
@@ -0,0 +1,175 @@
+//+build linux,lxc
+
+package lxc
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/plugins/drivers"
+	lxc "gopkg.in/lxc/go-lxc.v2"
+)
+
+var (
+	verbosityLevels = map[string]lxc.Verbosity{
+		"":        lxc.Quiet,
+		"verbose": lxc.Verbose,
+		"quiet":   lxc.Quiet,
+	}
+
+	logLevels = map[string]lxc.LogLevel{
+		"":      lxc.ERROR,
+		"debug": lxc.DEBUG,
+		"error": lxc.ERROR,
+		"info":  lxc.INFO,
+		"trace": lxc.TRACE,
+		"warn":  lxc.WARN,
+	}
+)
+
+const (
+	// containerMonitorIntv is the interval at which the driver checks if the
+	// container is still alive
+	containerMonitorIntv = 2 * time.Second
+)
+
+func (d *Driver) lxcPath() string {
+	lxcPath := d.config.Path
+	if lxcPath == "" {
+		lxcPath = lxc.DefaultConfigPath()
+	}
+	return lxcPath
+
+}
+func (d *Driver) initializeContainer(cfg *drivers.TaskConfig, taskConfig TaskConfig) (*lxc.Container, error) {
+	d.logger.Warn("initializing container", "lxc_path", d.config.Path, "default_path", lxc.DefaultConfigPath())
+	lxcPath := d.lxcPath()
+
+	containerName := fmt.Sprintf("%s-%s", cfg.Name, uuid.Generate())
```

Comment:
Using a UUID here.  Previous implementation used `alloc` id.  Considered using task id, but that adds escaping requirements (e.g. length, special characters, etc) and I think that strict known acceptable format is easier to reason about and better for future proofing.

Reply:
We might want to use taskid purely for consistency with the docker driver (https://github.com/hashicorp/nomad/blob/master/drivers/docker/driver.go#L877) - but then we'll also need to duplicate the dupe detection

Reply:
Just wanted to write down what we discussed earlier today:

We'll be plumbing in alloc id in #4891, so once that's merged let's make sure to switch to using it (either updating this PR, rebasing, another PR, etc. Whatever's easiest).

Reply:
@schmichael , @notnoop : #4891 is merged

Reply:
@cgbaker thanks for the update - using alloc ids now :).